### PR TITLE
Translate Markdown documentation to rST and remove recommonmark

### LINKED ISSUE #8636: [FEA] Use MyST for cudf markdown docs
Currently, cudf python rely on [recommonmark](https://github.com/readthedocs/recommonmark) to bridge markdown docs. However, this extension is deprecated. The suggested replacement is [MyST](https://github.com/executablebooks/MyST-Parser) parser.

### PR DESCRIPTION
**Pull request by vyasr**
Resolves #8636

### CODE REVIEW

**Review by ajschmidt8 — APPROVED** [review]
Approving `ops-codeowner` file changes

File: docs/cudf/source/groupby.rst [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+   small number of large groups, it can give acceptable performance
+
+-  The results may not always match Pandas exactly. For example, cuDF
+   may return a ``DataFrame`` containing a single column where Pandas
+   returns a ``Series``. Some post-processing may be required to match
+   Pandas behavior.
+
+-  cuDF does not support some of the exceptional cases that Pandas
+   supports with ``apply``, such as
+   ```describe`` <https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html#flexible-apply>`__.
```
Comment (isVoid) [inline docs/cudf/source/groupby.rst:205]: This doesn't render properly in sphinx. Tried a bit but sill not sure how to create a link for inline code. We could fall back to just regular links:
```suggestion
   `describe <https://pandas.pydata.org/pandas-docs/stable/user_guide/groupby.html#flexible-apply>`_.
```

File: docs/cudf/source/internals.rst [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+``['do', 'you', 'have', 'any', 'cheese?']`` is composed of:
+
+1. No data buffer
+2. No mask buffer as there are no nulls in the Series
+3. Two children columns:
+
+-  A column of 8-bit characters
+   ``['d', 'o', 'y', 'o', 'u', h' ... '?']``
+-  A column of "offsets" to the characters column (in this case,
+   ``[0, 2, 5, 9, 12, 19]``)
```
Comment (isVoid) [inline docs/cudf/source/internals.rst:60]: I think these two lines should be nested:
```suggestion
3. Two children columns:

    *  A column of 8-bit characters
    ``['d', 'o', 'y', 'o', 'u', h' ... '?']``

    *  A column of "offsets" to the characters column (in this case,
    ``[0, 2, 5, 9, 12, 19]``)
```

**Review by isVoid — COMMENTED** [review]
No obvious visual degradation compared to current page except the following two spots. In fact, this PR fixes a bunch of broken list of current page. E.g.:

Current:
image

This PR:
image

**Review by shwina — APPROVED** [review]

File: docs/cudf/source/groupby.rst (hunk 1 above)
Comment (vyasr) [reply docs/cudf/source/groupby.rst:205]: Good catch, I didn't notice this. I found a way to get it working.

**Review by isVoid — APPROVED** [review]

**Comment by vyasr** [comment]
@gpucibot merge

**Pull request merged**